packages/client/hmi-client/src/components/drilldown/tera-output-dropdown.vue

```diff
@@ -29,16 +29,29 @@
 </template>
 
 <script setup lang="ts">
+import { watch } from 'vue';
+import { isEmpty } from 'lodash';
 import { WorkflowOutput, WorkflowPortStatus } from '@/types/workflow';
 import Dropdown from 'primevue/dropdown';
 
-defineProps<{
+const props = defineProps<{
 	options: WorkflowOutput<any>[] | { label: string; items: WorkflowOutput<any>[] }[];
 	output: WorkflowOutput<any>['id'];
 	isLoading?: boolean;
 }>();
 
 const emit = defineEmits(['update:selection']);
+
+// When a new output is added to the list choose it by default
+watch(
+	() => props.options,
+	(newOptions) => {
+		if (!isEmpty(newOptions)) {
+			const { items } = newOptions[0];
+			emit('update:selection', items[items.length - 1].id);
+		}
+	}
+);
```

Comment:
Looks a little suspicious - let's have a sync on what are the symptoms and problems. 

As long as you invoke `append-output`, thing in theory should take care of itself, and while append-output can be triggered from update:selection, I don't think this is something you'd want to automatically invoke.